### DISCUSSION TITLE
Schema organization in the context of using Schema Stitching directives

### DISCUSSION THREAD

**Discussion by Drilmo** [ardatan/schema-stitching#1090] [Q&A]
hello, I use the "schema stitching" part with a gateway and to simplify maintenance and upgrades, I use "stitching directives" in the sub-services to make the links between them. 
However, I was wondering if it would be possible to better organize the graph as a whole (all the queries are scattered throughout the documentation, which makes reading a bit complex). 

Isn't it possible to use schemas like: 
```JavaScript
type typeA {
    id: ID!
    name: String!
  }
type Query {
 subServiceA: subServiceAQuery
}
type subServiceAQuery {
 "hello world"
 get(id: ID!): typeA @merge(keyField: "id")
}
```

I'm asking because I can't figure out how to do it, knowing that @merge can only be used as a root query field.  How can I do this?

**Comment by ardatan** [comment]
The same priniciples for schema modularization applies for services. So you can split parts of the schema in a different files;
https://the-guild.dev/graphql/tools/docs/schema-merging

About `@merge` directive, yes it can be only used on root fields because in GraphQL you can't query nested fields without starting from the root field. So it must be always root field so stitching can send requests for that specific root field.
